Fetch forward is none and returns “ERROR:  Index: 0”

### Which version of ShardingSphere did you use?
we find java version: java8, full_version=1.8.0_282, full_path=/usr/lib/jvm/java-1.8.0-openjdk-1.8.0.282.b08-1.el7_9.x86_64/bin/java
ShardingSphere-5.2.1-SNAPSHOT
Commit ID: dirty-a467bbe1668c2c300d3c91236e9e213c565ab7bf
Commit Message: Fix openGauss cursor execution in xa transaction(#21250) (#21257)
Branch: a467bbe1668c2c300d3c91236e9e213c565ab7bf
Build time: 2022-09-29T16:40:08+0800

### Which project did you use? ShardingSphere-JDBC or ShardingSphere-Proxy?
ShardingSphere-Proxy

### Expected behavior
Fetch forward is none and returns normal；
### Actual behavior
Fetch forward is none and returns “ERROR:  Index: 0”

```
test_db=> start transaction;
START TRANSACTION
test_db=> cursor cur_test_01 for select * from t_ssdb order by 1;
DECLARE CURSOR
test_db=> ----without forward
test_db=> fetch from cur_test_01;
 id | c_id | a  |   b
----+------+----+-------
  1 |    1 | 10 | test1
(1 row)

test_db=> ----forward without count
test_db=> fetch forward from cur_test_01;
 id | c_id | a  |   b
----+------+----+--------
  1 |    2 | 10 | Test12
(1 row)

test_db=> ----forward with count
test_db=> fetch forward 2 from cur_test_01;
 id | c_id | a  |   b
----+------+----+--------
  1 |    2 |    | test12
  2 |    2 | 20 | test22
(2 rows)

test_db=> ----forward with all
test_db=> fetch forward all from cur_test_01;
 id | c_id |  a  |    b
----+------+-----+---------
  2 |    1 |  20 | test21
  2 |    3 |     | Test22
  3 |    3 |     | test3
  3 |    3 |     | test3
  3 |    1 |     | test31
  4 |    4 |  40 | test4
  5 |    5 |  50 | Test5
  6 |    6 |  60 | test6
  7 |    4 |  74 | test74
  8 |    5 |  85 | test85
  9 |    5 |  50 | test95
 10 |    6 | 106 | test106
(12 rows)

test_db=> ----forward next is none
test_db=> fetch forward cur_test_01;
ERROR:  Index: 0
test_db=> close cur_test_01;
CLOSE CURSOR
test_db=> end;
COMMIT
```

background log

```
[INFO ] 2022-10-09 16:16:55.471 [Connection-56-ThreadExecutor] ShardingSphere-SQL - Actual SQL: ds_0 ::: fetch  FORWARD 1  cur_test_01_t_ssdb;
[INFO ] 2022-10-09 16:16:55.471 [Connection-56-ThreadExecutor] ShardingSphere-SQL - Actual SQL: ds_1 ::: fetch  FORWARD 1  cur_test_01_t_ssdb;
[ERROR] 2022-10-09 16:16:55.475 [Connection-56-ThreadExecutor] o.a.s.p.f.c.CommandExecutorTask - Exception occur:
java.lang.IndexOutOfBoundsException: Index: 0
        at java.util.Collections$EmptyList.get(Collections.java:4456)
        at org.apache.shardingsphere.sharding.merge.dql.orderby.OrderByValue.compareTo(OrderByValue.java:120)
        at org.apache.shardingsphere.sharding.merge.dql.orderby.OrderByValue.compareTo(OrderByValue.java:43)
        at java.util.PriorityQueue.siftUpComparable(PriorityQueue.java:656)
        at java.util.PriorityQueue.siftUp(PriorityQueue.java:647)
        at java.util.PriorityQueue.offer(PriorityQueue.java:344)
        at org.apache.shardingsphere.sharding.merge.ddl.fetch.FetchStreamMergedResult.addOrderedResultSetsToQueue(FetchStreamMergedResult.java:133)
        at org.apache.shardingsphere.sharding.merge.ddl.fetch.FetchStreamMergedResult.<init>(FetchStreamMergedResult.java:66)
        at org.apache.shardingsphere.sharding.merge.ddl.ShardingDDLResultMerger.merge(ShardingDDLResultMerger.java:58)
        at org.apache.shardingsphere.infra.merge.MergeEngine.executeMerge(MergeEngine.java:82)
        at org.apache.shardingsphere.infra.merge.MergeEngine.merge(MergeEngine.java:71)
        at org.apache.shardingsphere.proxy.backend.communication.DatabaseCommunicationEngine.mergeQuery(DatabaseCommunicationEngine.java:174)
        at org.apache.shardingsphere.proxy.backend.communication.DatabaseCommunicationEngine.processExecuteQuery(DatabaseCommunicationEngine.java:140)
        at org.apache.shardingsphere.proxy.backend.communication.jdbc.JDBCDatabaseCommunicationEngine.execute(JDBCDatabaseCommunicationEngine.java:134)
        at org.apache.shardingsphere.proxy.frontend.opengauss.command.query.simple.OpenGaussComQueryExecutor.execute(OpenGaussComQueryExecutor.java:76)
        at org.apache.shardingsphere.proxy.frontend.command.CommandExecutorTask.executeCommand(CommandExecutorTask.java:111)
        at org.apache.shardingsphere.proxy.frontend.command.CommandExecutorTask.run(CommandExecutorTask.java:78)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
```

### Reason analyze (If you can)

### Steps to reproduce the behavior, such as: SQL to execute, sharding rule configuration, when exception occur etc.
```
--create table
drop table if exists t_ssdb;
create table t_ssdb(id int,c_id int,a int,b text);
insert into t_ssdb values (1,1,10,'test1'),(1,2,10,'Test12'),(1,2,null,'test12'),
(2,1,20,'test21'),(2,2,20,'test22'),(2,3,null,'Test22'),
(3,3,null,'test3'),(3,3,null,'test3'),(3,1,null,'test31'),(4,4,40,'test4'),(5,5,50,'Test5'),(6,6,60,'test6'),
(7,4,74,'test74'),(8,5,85,'test85'),(9,5,50,'test95'),(10,6,106,'test106');

start transaction;
cursor cur_test_01 for select * from t_ssdb order by 1;
----without forward
fetch from cur_test_01;
----forward without count 
fetch forward from cur_test_01;
----forward with count
fetch forward 2 from cur_test_01;
----forward with all
fetch forward all from cur_test_01;
----forward next is none
fetch forward cur_test_01;
close cur_test_01;
end;
```

### Example codes for reproduce this issue (such as a github link).
```
rules:
- !SHARDING
  tables:
    t_ssdb:
      actualDataNodes: ds_${0..1}.t_ssdb
      databaseStrategy:
        standard:
          shardingAlgorithmName: ds_database_inline
          shardingColumn: id
  defaultTableStrategy:
    none:
  defaultDatabaseStrategy:
    none:
  shardingAlgorithms:
    ds_database_inline:
      type: INLINE
      props:
        algorithm-expression: ds_${id % 2}
        allow-range-query-with-inline-sharding: true
```

```
[INFO ] 2022-11-15 14:57:58.220 [Connection-7-ThreadExecutor] ShardingSphere-SQL - Actual SQL: write_0 ::: fetch FORWARD ALL from cur_test_01_t_ssdb;
[INFO ] 2022-11-15 14:57:58.220 [Connection-7-ThreadExecutor] ShardingSphere-SQL - Actual SQL: write_1 ::: fetch FORWARD ALL from cur_test_01_t_ssdb;
[INFO ] 2022-11-15 14:57:58.236 [Connection-7-ThreadExecutor] ShardingSphere-SQL - Logic SQL: fetch forward cur_test_01;
[INFO ] 2022-11-15 14:57:58.236 [Connection-7-ThreadExecutor] ShardingSphere-SQL - SQLStatement: OpenGaussFetchStatement(super=FetchStatement(super=AbstractSQLStatement(parameterCount=0, parameterMarkerSegments=[], commentSegments=[]), cursorName=CursorNameSegment(startIndex=14, stopIndex=24, identifier=IdentifierValue(value=cur_test_01, quoteCharacter=NONE)), direction=Optional[DirectionSegment(startIndex=6, stopIndex=12, directionType=Optional[FORWARD], count=Optional.empty)]))
[INFO ] 2022-11-15 14:57:58.236 [Connection-7-ThreadExecutor] ShardingSphere-SQL - Actual SQL: write_0 ::: fetch  FORWARD 1  cur_test_01_t_ssdb;
[INFO ] 2022-11-15 14:57:58.236 [Connection-7-ThreadExecutor] ShardingSphere-SQL - Actual SQL: write_1 ::: fetch  FORWARD 1  cur_test_01_t_ssdb;
[ERROR] 2022-11-15 14:57:58.240 [Connection-7-ThreadExecutor] o.a.s.p.f.c.CommandExecutorTask - Exception occur:
java.lang.IndexOutOfBoundsException: Index: 0
        at java.util.Collections$EmptyList.get(Collections.java:4456)
        at org.apache.shardingsphere.sharding.merge.dql.orderby.OrderByValue.compareTo(OrderByValue.java:120)
        at org.apache.shardingsphere.sharding.merge.dql.orderby.OrderByValue.compareTo(OrderByValue.java:43)
        at java.util.PriorityQueue.siftUpComparable(PriorityQueue.java:656)
        at java.util.PriorityQueue.siftUp(PriorityQueue.java:647)
        at java.util.PriorityQueue.offer(PriorityQueue.java:344)
        at org.apache.shardingsphere.sharding.merge.ddl.fetch.FetchStreamMergedResult.addOrderedResultSetsToQueue(FetchStreamMergedResult.java:133)
        at org.apache.shardingsphere.sharding.merge.ddl.fetch.FetchStreamMergedResult.<init>(FetchStreamMergedResult.java:66)
        at org.apache.shardingsphere.sharding.merge.ddl.ShardingDDLResultMerger.merge(ShardingDDLResultMerger.java:58)
        at org.apache.shardingsphere.infra.merge.MergeEngine.executeMerge(MergeEngine.java:82)
        at org.apache.shardingsphere.infra.merge.MergeEngine.merge(MergeEngine.java:71)
        at org.apache.shardingsphere.proxy.backend.communication.DatabaseCommunicationEngine.mergeQuery(DatabaseCommunicationEngine.java:174)
        at org.apache.shardingsphere.proxy.backend.communication.DatabaseCommunicationEngine.processExecuteQuery(DatabaseCommunicationEngine.java:140)
        at org.apache.shardingsphere.proxy.backend.communication.jdbc.JDBCDatabaseCommunicationEngine.execute(JDBCDatabaseCommunicationEngine.java:134)
        at org.apache.shardingsphere.proxy.frontend.opengauss.command.query.simple.OpenGaussComQueryExecutor.execute(OpenGaussComQueryExecutor.java:76)
        at org.apache.shardingsphere.proxy.frontend.command.CommandExecutorTask.executeCommand(CommandExecutorTask.java:111)
        at org.apache.shardingsphere.proxy.frontend.command.CommandExecutorTask.run(CommandExecutorTask.java:78)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
[INFO ] 2022-11-15 14:57:58.242 [Connection-7-ThreadExecutor] ShardingSphere-SQL - Logic SQL: close cur_test_01;
[INFO ] 2022-11-15 14:57:58.242 [Connection-7-ThreadExecutor] ShardingSphere-SQL - SQLStatement: OpenGaussCloseStatement(super=CloseStatement(super=AbstractSQLStatement(parameterCount=0, parameterMarkerSegments=[], commentSegments=[]), cursorName=Optional[CursorNameSegment(startIndex=6, stopIndex=16, identifier=IdentifierValue(value=cur_test_01, quoteCharacter=NONE))], closeAll=false))
[INFO ] 2022-11-15 14:57:58.242 [Connection-7-ThreadExecutor] ShardingSphere-SQL - Actual SQL: write_0 ::: close cur_test_01_t_ssdb;
[INFO ] 2022-11-15 14:57:58.242 [Connection-7-ThreadExecutor] ShardingSphere-SQL - Actual SQL: write_1 ::: close cur_test_01_t_ssdb;
```



```
[peilq_sharding@kwemhisprc10431 auto_test]$ bash ss_home/bin/start.sh --version
we find java version: java8, full_version=1.8.0_282, full_path=/usr/lib/jvm/java-1.8.0-openjdk-1.8.0.282.b08-1.el7_9.x86_64/bin/java
ShardingSphere-5.2.1-SNAPSHOT
Commit ID: dirty-20bf595dfeced4dd8ffee2f6d95de52fdf3e569d
Commit Message: Rename sub modules of shardingsphere-infra-datetime-type. (#21472)
Branch: 20bf595dfeced4dd8ffee2f6d95de52fdf3e569d
Build time: 2022-11-11T19:57:07+0800

```

--- a/features/sharding/core/src/main/java/org/apache/shardingsphere/sharding/merge/ddl/fetch/FetchStreamMergedResult.java
+++ b/features/sharding/core/src/main/java/org/apache/shardingsphere/sharding/merge/ddl/fetch/FetchStreamMergedResult.java
@@ -133,8 +133,7 @@ public final class FetchStreamMergedResult extends StreamMergedResult {
     }
     
     private boolean isEmptyOrderByValue(final OrderByValue orderByValue) {
-        return orderByValue.getQueryResult() instanceof JDBCMemoryQueryResult
-                && 0 == ((JDBCMemoryQueryResult) orderByValue.getQueryResult()).getRowCount() && ((JDBCMemoryQueryResult) orderByValue.getQueryResult()).wasNull();
+        return orderByValue.getQueryResult() instanceof JDBCMemoryQueryResult && 0 == ((JDBCMemoryQueryResult) orderByValue.getQueryResult()).getRowCount();
     }
     
     private void addOrderedResultSetsToQueue(final List<FetchOrderByValueGroup> fetchOrderByValueGroups, final List<QueryResult> queryResults) {

--- a/test/integration-test/transaction/src/test/java/org/apache/shardingsphere/integration/transaction/cases/cursor/OpenGaussCursorTestCase.java
+++ b/test/integration-test/transaction/src/test/java/org/apache/shardingsphere/integration/transaction/cases/cursor/OpenGaussCursorTestCase.java
@@ -66,7 +66,7 @@ public final class OpenGaussCursorTestCase extends BaseTransactionTestCase {
         broadcastAndSingleTablesCursorTest2(connection);
         viewCursorTest(connection);
     }
-
+    
     private void singleTableCursorTest(final Connection connection) throws SQLException {
         executeWithLog(connection, "start transaction;");
         executeWithLog(connection, cursorSQLCommand.getSingleTableCursor());
@@ -76,28 +76,28 @@ public final class OpenGaussCursorTestCase extends BaseTransactionTestCase {
         fetch(connection, 2);
         fetch(connection, 3);
         fetch(connection, 4);
+        fetchOverTest(connection);
+        executeWithLog(connection, "rollback;");
+    }
+    
+    private void fetchOverTest(final Connection connection) throws SQLException {
         fetchOver(connection);
         fetchOver(connection);
         fetchForwardOver(connection);
         fetchForwardAllOver(connection);
         fetchForwardOver(connection);
-        executeWithLog(connection, "rollback;");
     }
-
+    
     private void singleTableCursorOrderByTest(final Connection connection) throws SQLException {
         executeWithLog(connection, "start transaction;");
-        executeWithLog(connection, cursorSQLCommand.getSingleTableCursor());
+        executeWithLog(connection, cursorSQLCommand.getSingleTableCursorOrderBy());
         executeWithLog(connection, "close test;");
-        executeWithLog(connection, cursorSQLCommand.getSingleTableCursor());
+        executeWithLog(connection, cursorSQLCommand.getSingleTableCursorOrderBy());
         fetch(connection, 4);
         fetch(connection, 3);
         fetch(connection, 2);
         fetch(connection, 1);
-        fetchOver(connection);
-        fetchOver(connection);
-        fetchForwardOver(connection);
-        fetchForwardAllOver(connection);
-        fetchForwardOver(connection);
+        fetchOverTest(connection);
         executeWithLog(connection, "rollback;");
     }
     
@@ -110,11 +110,7 @@ public final class OpenGaussCursorTestCase extends BaseTransactionTestCase {
         fetch(connection, 10102);
         fetch(connection, 10201);
         fetch(connection, 10202);
-        fetchOver(connection);
-        fetchOver(connection);
-        fetchForwardOver(connection);
-        fetchForwardAllOver(connection);
-        fetchForwardOver(connection);
+        fetchOverTest(connection);
         executeWithLog(connection, "rollback;");
     }
     
@@ -127,11 +123,7 @@ public final class OpenGaussCursorTestCase extends BaseTransactionTestCase {
         fetch(connection, 10102);
         fetch(connection, 10201);
         fetch(connection, 10202);
-        fetchOver(connection);
-        fetchOver(connection);
-        fetchForwardOver(connection);
-        fetchForwardAllOver(connection);
-        fetchForwardOver(connection);
+        fetchOverTest(connection);
         executeWithLog(connection, "rollback;");
     }
     
@@ -144,11 +136,7 @@ public final class OpenGaussCursorTestCase extends BaseTransactionTestCase {
         fetch(connection, 2);
         fetch(connection, 3);
         fetch(connection, 4);
-        fetchOver(connection);
-        fetchOver(connection);
-        fetchForwardOver(connection);
-        fetchForwardAllOver(connection);
-        fetchForwardOver(connection);
+        fetchOverTest(connection);
         executeWithLog(connection, "rollback;");
     }
     
@@ -161,8 +149,7 @@ public final class OpenGaussCursorTestCase extends BaseTransactionTestCase {
         fetch(connection, 10102);
         fetch(connection, 10201);
         fetch(connection, 10202);
-        fetchOver(connection);
-        fetchOver(connection);
+        fetchOverTest(connection);
         executeWithLog(connection, "rollback;");
     }
     
@@ -175,11 +162,7 @@ public final class OpenGaussCursorTestCase extends BaseTransactionTestCase {
         fetch(connection, 2);
         fetch(connection, 3);
         fetch(connection, 4);
-        fetchOver(connection);
-        fetchOver(connection);
-        fetchForwardOver(connection);
-        fetchForwardAllOver(connection);
-        fetchForwardOver(connection);
+        fetchOverTest(connection);
         executeWithLog(connection, "rollback;");
     }
     

--- a/test/integration-test/transaction/src/test/java/org/apache/shardingsphere/integration/transaction/cases/cursor/OpenGaussCursorTestCase.java
+++ b/test/integration-test/transaction/src/test/java/org/apache/shardingsphere/integration/transaction/cases/cursor/OpenGaussCursorTestCase.java
@@ -58,11 +58,47 @@ public final class OpenGaussCursorTestCase extends BaseTransactionTestCase {
     @Override
     public void executeTest() throws SQLException {
         Connection connection = getDataSource().getConnection();
+        singleTableCursorTest(connection);
+        singleTableCursorOrderByTest(connection);
         broadcastTableCursorTest(connection);
         broadcastTableCursorTest2(connection);
         broadcastAndSingleTablesCursorTest(connection);
         broadcastAndSingleTablesCursorTest2(connection);
         viewCursorTest(connection);
+    }
+
+    private void singleTableCursorTest(final Connection connection) throws SQLException {
+        executeWithLog(connection, "start transaction;");
+        executeWithLog(connection, cursorSQLCommand.getSingleTableCursor());
+        executeWithLog(connection, "close test;");
+        executeWithLog(connection, cursorSQLCommand.getSingleTableCursor());
+        fetch(connection, 1);
+        fetch(connection, 2);
+        fetch(connection, 3);
+        fetch(connection, 4);
+        fetchOver(connection);
+        fetchOver(connection);
+        fetchForwardOver(connection);
+        fetchForwardAllOver(connection);
+        fetchForwardOver(connection);
+        executeWithLog(connection, "rollback;");
+    }
+
+    private void singleTableCursorOrderByTest(final Connection connection) throws SQLException {
+        executeWithLog(connection, "start transaction;");
+        executeWithLog(connection, cursorSQLCommand.getSingleTableCursor());
+        executeWithLog(connection, "close test;");
+        executeWithLog(connection, cursorSQLCommand.getSingleTableCursor());
+        fetch(connection, 4);
+        fetch(connection, 3);
+        fetch(connection, 2);
+        fetch(connection, 1);
+        fetchOver(connection);
+        fetchOver(connection);
+        fetchForwardOver(connection);
+        fetchForwardAllOver(connection);
+        fetchForwardOver(connection);
+        executeWithLog(connection, "rollback;");
     }
     
     private void broadcastTableCursorTest(final Connection connection) throws SQLException {
@@ -76,6 +112,9 @@ public final class OpenGaussCursorTestCase extends BaseTransactionTestCase {
         fetch(connection, 10202);
         fetchOver(connection);
         fetchOver(connection);
+        fetchForwardOver(connection);
+        fetchForwardAllOver(connection);
+        fetchForwardOver(connection);
         executeWithLog(connection, "rollback;");
     }
     
@@ -90,6 +129,9 @@ public final class OpenGaussCursorTestCase extends BaseTransactionTestCase {
         fetch(connection, 10202);
         fetchOver(connection);
         fetchOver(connection);
+        fetchForwardOver(connection);
+        fetchForwardAllOver(connection);
+        fetchForwardOver(connection);
         executeWithLog(connection, "rollback;");
     }
     
@@ -104,6 +146,9 @@ public final class OpenGaussCursorTestCase extends BaseTransactionTestCase {
         fetch(connection, 4);
         fetchOver(connection);
         fetchOver(connection);
+        fetchForwardOver(connection);
+        fetchForwardAllOver(connection);
+        fetchForwardOver(connection);
         executeWithLog(connection, "rollback;");
     }
     
@@ -132,6 +177,9 @@ public final class OpenGaussCursorTestCase extends BaseTransactionTestCase {
         fetch(connection, 4);
         fetchOver(connection);
         fetchOver(connection);
+        fetchForwardOver(connection);
+        fetchForwardAllOver(connection);
+        fetchForwardOver(connection);
         executeWithLog(connection, "rollback;");
     }
     
@@ -145,6 +193,20 @@ public final class OpenGaussCursorTestCase extends BaseTransactionTestCase {
     
     private void fetchOver(final Connection connection) throws SQLException {
         ResultSet resultSet = executeQueryWithLog(connection, "fetch test;");
+        while (resultSet.next()) {
+            fail("Expected fetch nothing.");
+        }
+    }
+    
+    private void fetchForwardOver(final Connection connection) throws SQLException {
+        ResultSet resultSet = executeQueryWithLog(connection, "fetch forward from test;");
+        while (resultSet.next()) {
+            fail("Expected fetch nothing.");
+        }
+    }
+    
+    private void fetchForwardAllOver(final Connection connection) throws SQLException {
+        ResultSet resultSet = executeQueryWithLog(connection, "fetch forward all from test;");
         while (resultSet.next()) {
             fail("Expected fetch nothing.");
         }

--- a/test/integration-test/transaction/src/test/java/org/apache/shardingsphere/integration/transaction/engine/command/CursorSQLCommand.java
+++ b/test/integration-test/transaction/src/test/java/org/apache/shardingsphere/integration/transaction/engine/command/CursorSQLCommand.java
@@ -31,6 +31,12 @@ import javax.xml.bind.annotation.XmlRootElement;
 @Setter
 public final class CursorSQLCommand {
     
+    @XmlElement(name = "single-table-cursor")
+    private String singleTableCursor;
+    
+    @XmlElement(name = "single-table-cursor-order-by")
+    private String singleTableCursorOrderBy;
+    
     @XmlElement(name = "broadcast-tables-cursor")
     private String broadcastTablesCursor;
     

--- a/test/integration-test/transaction/src/test/resources/env/common/cursor-command.xml
+++ b/test/integration-test/transaction/src/test/resources/env/common/cursor-command.xml
@@ -16,6 +16,17 @@
   -->
 
 <command>
+    <single-table-cursor>
+        cursor test for
+        select * from t_product;
+    </single-table-cursor>
+    
+    <single-table-cursor-order-by>
+        cursor test for
+        select * from t_product
+        order by product_id desc;
+    </single-table-cursor-order-by>
+    
     <broadcast-tables-cursor>
         cursor test for
         select t_province.province_id, t_province.province_name, t_city.city_id as id, t_city.city_name

--- a/test/integration-test/transaction/src/test/resources/env/scenario/cursor/data/actual/init-sql/opengauss/01-actual-init.sql
+++ b/test/integration-test/transaction/src/test/resources/env/scenario/cursor/data/actual/init-sql/opengauss/01-actual-init.sql
@@ -40,7 +40,7 @@ GRANT ALL PRIVILEGES ON DATABASE transaction_it_2 TO test_user;
 
 \c transaction_it_0;
 
-CREATE TABLE "t_product" ("product_id" INT NOT NULL, "address_id" INT, "product_name" varchar, "category_id" INT NOT NULL, "price" numeric NOT NULL, "status" varchar, "creation_date" date);
+CREATE TABLE "t_product" ("id" INT PRIMARY KEY, "product_id" INT NOT NULL, "address_id" INT, "product_name" varchar, "category_id" INT NOT NULL, "price" numeric NOT NULL, "status" varchar, "creation_date" date);
 CREATE TABLE "t_order_0" ("order_id" INT PRIMARY KEY, "address_id" INT, "user_id" INT NOT NULL, "status" VARCHAR(45) NULL);
 CREATE TABLE "t_order_1" ("order_id" INT PRIMARY KEY, "address_id" INT, "user_id" INT NOT NULL, "status" VARCHAR(45) NULL);
 CREATE TABLE "t_order_item_0" ("item_id" INT PRIMARY KEY, "order_id" int NOT NULL, "user_id" int NOT NULL, "status" varchar(50) DEFAULT NULL);
@@ -72,11 +72,12 @@ INSERT INTO "t_country" ("country_id", "country_name", "continent_name", "creati
 INSERT INTO "t_order_0" ("order_id", "address_id", "user_id", "status") VALUES (2, 1, 2, 'OK');
 INSERT INTO "t_order_1" ("order_id", "address_id", "user_id", "status") VALUES (1, 2, 2, 'OK');
 
-INSERT INTO "t_product" ("product_id", "address_id", "product_name", "category_id", "price", "status", "creation_date") VALUES (1, 1, 'product1', 1, 1, 'OK', '2022-11-02');
-INSERT INTO "t_product" ("product_id", "address_id", "product_name", "category_id", "price", "status", "creation_date") VALUES (2, 2, 'product2', 2, 2, 'OK', '2022-11-02');
-INSERT INTO "t_product" ("product_id", "address_id", "product_name", "category_id", "price", "status", "creation_date") VALUES (3, 3, 'product3', 3, 3, 'OK', '2022-11-02');
-INSERT INTO "t_product" ("product_id", "address_id", "product_name", "category_id", "price", "status", "creation_date") VALUES (4, 4, 'product4', 4, 4, 'OK', '2022-11-02');
+INSERT INTO "t_product" ("id", "product_id", "address_id", "product_name", "category_id", "price", "status", "creation_date") VALUES (1, 1, 1, 'product1', 1, 1, 'OK', '2022-11-02');
+INSERT INTO "t_product" ("id", "product_id", "address_id", "product_name", "category_id", "price", "status", "creation_date") VALUES (2, 2, 2, 'product2', 2, 2, 'OK', '2022-11-02');
+INSERT INTO "t_product" ("id", "product_id", "address_id", "product_name", "category_id", "price", "status", "creation_date") VALUES (3, 3, 3, 'product3', 3, 3, 'OK', '2022-11-02');
+INSERT INTO "t_product" ("id", "product_id", "address_id", "product_name", "category_id", "price", "status", "creation_date") VALUES (4, 4, 4, 'product4', 4, 4, 'OK', '2022-11-02');
 
+\c
 \c transaction_it_1;
 
 CREATE TABLE "t_order_0" ("order_id" INT PRIMARY KEY, "address_id" INT, "user_id" INT NOT NULL, "status" VARCHAR(45) NULL);

